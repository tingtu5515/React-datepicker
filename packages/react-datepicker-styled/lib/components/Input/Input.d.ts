interface InputProps {
    placeholder: string;
    value: string;
    id: string;
    ariaLabel: string;
    onClick(): void;
}
declare function Input({ placeholder, id, ariaLabel, onClick, value }: InputProps): JSX.Element;
export default Input;
