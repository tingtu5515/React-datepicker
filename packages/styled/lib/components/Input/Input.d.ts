import {ResponsiveValue, TLengthStyledSystem} from 'styled-system'
import {PaddingProperty} from 'csstype'
interface InputProps {
  placeholder: string
  value: string
  id: string
  ariaLabel: string
  onClick(): void
  showCalendarIcon: boolean
  vertical: boolean
  isActive: boolean
  padding?: ResponsiveValue<PaddingProperty<TLengthStyledSystem>>
}
declare function Input({
  placeholder,
  id,
  vertical,
  isActive,
  ariaLabel,
  onClick,
  value,
  showCalendarIcon,
  padding,
}: InputProps): JSX.Element
export default Input
