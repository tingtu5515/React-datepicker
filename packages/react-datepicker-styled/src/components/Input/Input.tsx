import React from 'react'
import styled from 'styled-components'
import {
  background,
  BackgroundProps,
  border,
  BorderProps,
  borderRadius,
  BorderRadiusProps,
  space,
  SpaceProps,
  fontFamily,
  FontFamilyProps,
  fontSize,
  FontSizeProps,
  color,
  ColorProps,
  fontWeight,
  FontWeightProps,
  position,
  PositionProps,
  top,
  TopProps,
  left,
  LeftProps,
  width,
  WidthProps,
  height,
  HeightProps,
  display,
  DisplayProps,
} from 'styled-system'
import CalendarIcon from '../../icons/CalendarIcon'

interface InputLabelProps
  extends PositionProps,
    BorderProps,
    BackgroundProps,
    DisplayProps,
    BorderRadiusProps {}
const InputLabel = styled('label')<InputLabelProps>`
  ${position}
  ${border}
  ${background}
  ${display}
  ${borderRadius}
`

interface CalendarWrapperProps
  extends PositionProps,
    LeftProps,
    TopProps,
    HeightProps,
    WidthProps {}
const CalendarWrapper = styled('div')<CalendarWrapperProps>`
  ${position}
  ${left}
  ${top}
  ${height}
  ${width}
  
  svg {
    display: block;
  }
`

interface StyledInputProps
  extends BackgroundProps,
    SpaceProps,
    FontFamilyProps,
    ColorProps,
    FontWeightProps,
    BorderProps,
    WidthProps,
    HeightProps,
    FontSizeProps {}
const StyledInput = styled('input')<StyledInputProps>`
  ${background}
  ${space}
  ${fontFamily}
  ${fontSize}
  ${color}
  ${fontWeight}
  ${space}
  ${border}
  ${width}
  ${height}
  cursor: pointer;
  box-sizing: border-box;
  
  ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
    font-weight: 500;
    color: #929598;
  }
  ::-moz-placeholder { /* Firefox 19+ */
    font-weight: 500;
    color: #929598;
  }
  :-moz-placeholder { /* Firefox 18- */
    font-weight: 500;
    color: #929598;
  }
`

function Input() {
  return (
    <InputLabel
      htmlFor="test"
      display="block"
      position="relative"
      border="1px solid #d0d0d0"
      background="#ffffff"
      borderRadius="2px"
    >
      <CalendarWrapper position="absolute" height="12px" width="12px" top="18px" left="16px">
        <CalendarIcon width="12px" height="12px" color="#BCBEC0" />
      </CalendarWrapper>
      <StyledInput
        border="0"
        px="44px"
        width="100%"
        height="46px"
        background="#ffffff"
        fontFamily="Montserrat"
        color="#001217"
        fontSize="14px"
        fontWeight={600}
        id="test"
        placeholder="Set end date"
        readOnly
      />
    </InputLabel>
  )
}

export default Input
