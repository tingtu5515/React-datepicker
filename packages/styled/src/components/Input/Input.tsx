import React from 'react'
import styled from 'styled-components'
import {
  background,
  BackgroundProps,
  border,
  BorderProps,
  borderRadius,
  BorderRadiusProps,
  space,
  SpaceProps,
  fontFamily,
  FontFamilyProps,
  fontSize,
  FontSizeProps,
  color,
  ColorProps,
  fontWeight,
  FontWeightProps,
  position,
  PositionProps,
  top,
  TopProps,
  left,
  LeftProps,
  width,
  WidthProps,
  height,
  HeightProps,
  display,
  DisplayProps,
  minHeight,
  MinHeightProps,
  boxShadow,
  BoxShadowProps,
  right,
  RightProps,
  style,
  ResponsiveValue,
  TLengthStyledSystem,
} from 'styled-system'
import CalendarIcon from '../../icons/CalendarIcon'
// eslint-disable-next-line import/no-unresolved
import {InputTheme} from '../../@types/theme'
import useThemeProps from '../../hooks/useThemeProps'
import globalStyles from '../../globalStyles'
import {PaddingProperty} from 'csstype'

const placeholderColor = style({
  prop: 'placeholderColor',
  cssProperty: 'color',
})

const placeholderFontWeight = style({
  prop: 'placeholderFontWeight',
  cssProperty: 'fontWeight',
})

interface InputLabelProps
  extends PositionProps,
    BorderProps,
    BackgroundProps,
    DisplayProps,
    SpaceProps,
    BorderRadiusProps {}
const InputLabel = styled('label')<InputLabelProps>`
  ${position}
  ${border}
  ${background}
  ${display}
  ${borderRadius}
  ${space}
`

interface CalendarWrapperProps
  extends PositionProps,
    LeftProps,
    RightProps,
    TopProps,
    HeightProps,
    WidthProps {}
const CalendarWrapper = styled('div')<CalendarWrapperProps>`
  ${position}
  ${left}
  ${right}
  ${top}
  ${height}
  ${width}
  cursor: pointer;
  
  svg {
    display: block;
  }
`

interface StyledInputProps
  extends BackgroundProps,
    SpaceProps,
    FontFamilyProps,
    ColorProps,
    FontWeightProps,
    BorderProps,
    WidthProps,
    MinHeightProps,
    BoxShadowProps,
    FontSizeProps {}
const StyledInput = styled('input')<StyledInputProps>`
  ${background}
  ${space}
  ${fontFamily}
  ${fontSize}
  ${color}
  ${fontWeight}
  ${space}
  ${border}
  ${width}
  ${minHeight}
  ${boxShadow}
  cursor: pointer;
  box-sizing: border-box;
  outline: 0;
  
  ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
    ${placeholderFontWeight}
    ${placeholderColor}
  }
  ::-moz-placeholder { /* Firefox 19+ */
    ${placeholderFontWeight}
    ${placeholderColor}
  }
  :-moz-placeholder { /* Firefox 18- */
    ${placeholderFontWeight}
    ${placeholderColor}
  }
`

interface InputProps {
  placeholder: string
  value: string
  id: string
  ariaLabel: string
  onClick(): void
  showCalendarIcon: boolean
  vertical: boolean
  isActive: boolean
  rtl: boolean
  disableAccessibility?: boolean
  padding?: ResponsiveValue<PaddingProperty<TLengthStyledSystem>>
}

function Input({
  placeholder,
  id,
  vertical,
  isActive,
  ariaLabel,
  onClick,
  value,
  showCalendarIcon,
  padding,
  rtl,
  disableAccessibility,
}: InputProps) {
  const theme: InputTheme = useThemeProps({
    fontFamily: globalStyles.fontFamily,
    inputFontWeight: 600,
    inputFontSize: '14px',
    inputColor: globalStyles.colors.charcoal,
    inputBackground: '#ffffff',
    inputMinHeight: '46px',
    inputWidth: '100%',
    inputPadding: padding,
    inputBorder: '0',
    inputPlaceholderFontWeight: 500,
    inputPlaceholderColor: globalStyles.colors.silverCloud,
    inputCalendarWrapperPosition: 'absolute',
    inputCalendarWrapperHeight: '12px',
    inputCalendarWrapperWidth: '12px',
    inputCalendarWrapperTop: '16px',
    inputCalendarWrapperLeft: rtl ? 'unset' : vertical ? '8px' : '16px',
    inputCalendarWrapperRight: rtl ? (vertical ? '8px' : '16px') : 'unset',
    inputCalendarIconWidth: '12px',
    inputCalendarIconHeight: '12px',
    inputCalendarIconColor: '#BCBEC0',
    inputLabelDisplay: 'block',
    inputLabelPosition: 'relative',
    inputLabelBorder: '1px solid #d0d0d0',
    inputLabelBorderRadius: '2px',
    inputLabelBackground: '#ffffff',
    inputLabelMargin: '0',
    inputActiveBoxShadow: 'inset 0px -3px 0 #00aeef',
  })

  return (
    <InputLabel
      htmlFor={id}
      display={theme.inputLabelDisplay}
      position={theme.inputLabelPosition}
      border={theme.inputLabelBorder}
      background={theme.inputLabelBackground}
      borderRadius={theme.inputLabelBorderRadius}
      m={theme.inputLabelMargin}
    >
      {showCalendarIcon && (
        <CalendarWrapper
          position={theme.inputCalendarWrapperPosition}
          height={theme.inputCalendarWrapperHeight}
          width={theme.inputCalendarWrapperWidth}
          top={theme.inputCalendarWrapperTop}
          left={theme.inputCalendarWrapperLeft}
          right={theme.inputCalendarWrapperRight}
        >
          <CalendarIcon
            // @ts-ignore
            width={theme.inputCalendarIconWidth}
            // @ts-ignore
            height={theme.inputCalendarIconHeight}
            // @ts-ignore
            color={theme.inputCalendarIconColor}
          />
        </CalendarWrapper>
      )}
      <StyledInput
        readOnly
        tabIndex={disableAccessibility ? -1 : 0}
        border={theme.inputBorder}
        p={theme.inputPadding}
        // @ts-ignore
        width={theme.inputWidth}
        minHeight={theme.inputMinHeight}
        background={theme.inputBackground}
        fontFamily={theme.fontFamily}
        // @ts-ignore
        color={theme.inputColor}
        fontSize={theme.inputFontSize}
        fontWeight={theme.inputFontWeight}
        placeholderColor={theme.inputPlaceholderColor}
        placeholderFontWeight={theme.inputPlaceholderFontWeight}
        boxShadow={isActive ? theme.inputActiveBoxShadow : 'none'}
        id={id}
        placeholder={placeholder}
        aria-label={ariaLabel}
        value={value}
        autoComplete="off"
        onFocus={onClick}
        data-testid="DatepickerInput"
      />
    </InputLabel>
  )
}

export default Input
