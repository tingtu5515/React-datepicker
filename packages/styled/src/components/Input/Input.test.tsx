import * as React from 'react'
import {render, fireEvent} from 'react-testing-library'
import Input from '.'

test('should render with icon', () => {
  const onClick = jest.fn()
  const {container, getByPlaceholderText} = render(
    <Input
      rtl={false}
      vertical={false}
      isActive={false}
      showCalendarIcon
      ariaLabel="startDate"
      id="startDate"
      placeholder="Placeholder"
      onClick={onClick}
      value=""
    />,
  )

  expect(container).toMatchSnapshot()
  getByPlaceholderText('Placeholder')
  // @ts-ignore
  fireEvent.focus(getByPlaceholderText('Placeholder'))
  expect(onClick).toHaveBeenCalled()
})

test('should render rtl variant', () => {
  const onClick = jest.fn()
  const {container, getByPlaceholderText} = render(
    <Input
      rtl
      vertical={false}
      isActive={false}
      showCalendarIcon
      ariaLabel="startDate"
      id="startDate"
      placeholder="Placeholder"
      onClick={onClick}
      value=""
    />,
  )

  expect(container).toMatchSnapshot()
  getByPlaceholderText('Placeholder')
  // @ts-ignore
  fireEvent.focus(getByPlaceholderText('Placeholder'))
  expect(onClick).toHaveBeenCalled()
})

test('should render without icon', () => {
  const onClick = jest.fn()
  const {container} = render(
    <Input
      rtl={false}
      vertical={false}
      showCalendarIcon={false}
      isActive={false}
      ariaLabel="startDate"
      id="startDate"
      placeholder="Placeholder"
      onClick={onClick}
      value=""
    />,
  )

  expect(container).toMatchSnapshot()
})

test('should render with value', () => {
  const onClick = jest.fn()
  const {container, getByDisplayValue} = render(
    <Input
      rtl={false}
      vertical={false}
      showCalendarIcon={false}
      isActive={false}
      ariaLabel="startDate"
      id="startDate"
      placeholder="Placeholder"
      onClick={onClick}
      value="14/11/1992"
    />,
  )

  expect(container).toMatchSnapshot()
  getByDisplayValue('14/11/1992')
})

test('should render vertical variant', () => {
  const onClick = jest.fn()
  const {container} = render(
    <Input
      vertical
      isActive
      rtl={false}
      showCalendarIcon={false}
      ariaLabel="startDate"
      id="startDate"
      placeholder="Placeholder"
      onClick={onClick}
      value="14/11/1992"
    />,
  )

  expect(container).toMatchSnapshot()
})
