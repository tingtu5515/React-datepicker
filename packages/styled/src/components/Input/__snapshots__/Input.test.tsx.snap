// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render rtl variant 1`] = `
.c0 {
  position: relative;
  border: 1px solid #d0d0d0;
  background: #ffffff;
  display: block;
  border-radius: 2px;
  margin: 0;
}

.c1 {
  position: absolute;
  left: unset;
  right: 16px;
  top: 16px;
  height: 12px;
  width: 12px;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

.c2 {
  background: #ffffff;
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  color: #001217;
  font-weight: 600;
  border: 0;
  width: 100%;
  min-height: 46px;
  box-shadow: none;
  cursor: pointer;
  box-sizing: border-box;
  outline: 0;
}

.c2::-webkit-input-placeholder {
  font-weight: 500;
  color: #929598;
}

.c2::-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

.c2:-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

<div>
  <label
    class="c0"
    display="block"
    for="startDate"
  >
    <div
      class="c1"
      height="12px"
      width="12px"
    >
      <svg
        class=""
        color="#BCBEC0"
        height="12px"
        viewBox="0 0 12 12"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 1H7v1h1V1zM6.5 6.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM6 3a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-1 0v2A.5.5 0 0 0 6 3zm3.5 5.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zm0-2h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM9 3a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-1 0v2A.5.5 0 0 0 9 3zm-.5 2.5h1a.5.5 0 0 0 0-1h-1a.5.5 0 0 0 0 1zm-3 0h1a.5.5 0 0 0 0-1h-1a.5.5 0 0 0 0 1zm-2 3h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM11 1h-1v1h1v9H1V2h1V1H1a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1zM3.5 6.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM5 1H4v1h1V1zm1.5 7.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zm-4-3h1a.5.5 0 0 0 0-1h-1a.5.5 0 0 0 0 1zM3 3a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-1 0v2A.5.5 0 0 0 3 3z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </div>
    <input
      aria-label="startDate"
      autocomplete="off"
      class="c2"
      color="#001217"
      data-testid="DatepickerInput"
      font-family="Montserrat, sans-serif"
      font-size="14px"
      font-weight="600"
      id="startDate"
      placeholder="Placeholder"
      readonly=""
      tabindex="0"
      value=""
      width="100%"
    />
  </label>
</div>
`;

exports[`should render vertical variant 1`] = `
.c0 {
  position: relative;
  border: 1px solid #d0d0d0;
  background: #ffffff;
  display: block;
  border-radius: 2px;
  margin: 0;
}

.c1 {
  background: #ffffff;
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  color: #001217;
  font-weight: 600;
  border: 0;
  width: 100%;
  min-height: 46px;
  box-shadow: inset 0px -3px 0 #00aeef;
  cursor: pointer;
  box-sizing: border-box;
  outline: 0;
}

.c1::-webkit-input-placeholder {
  font-weight: 500;
  color: #929598;
}

.c1::-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

.c1:-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

<div>
  <label
    class="c0"
    display="block"
    for="startDate"
  >
    <input
      aria-label="startDate"
      autocomplete="off"
      class="c1"
      color="#001217"
      data-testid="DatepickerInput"
      font-family="Montserrat, sans-serif"
      font-size="14px"
      font-weight="600"
      id="startDate"
      placeholder="Placeholder"
      readonly=""
      tabindex="0"
      value="14/11/1992"
      width="100%"
    />
  </label>
</div>
`;

exports[`should render with icon 1`] = `
.c0 {
  position: relative;
  border: 1px solid #d0d0d0;
  background: #ffffff;
  display: block;
  border-radius: 2px;
  margin: 0;
}

.c1 {
  position: absolute;
  left: 16px;
  right: unset;
  top: 16px;
  height: 12px;
  width: 12px;
  cursor: pointer;
}

.c1 svg {
  display: block;
}

.c2 {
  background: #ffffff;
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  color: #001217;
  font-weight: 600;
  border: 0;
  width: 100%;
  min-height: 46px;
  box-shadow: none;
  cursor: pointer;
  box-sizing: border-box;
  outline: 0;
}

.c2::-webkit-input-placeholder {
  font-weight: 500;
  color: #929598;
}

.c2::-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

.c2:-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

<div>
  <label
    class="c0"
    display="block"
    for="startDate"
  >
    <div
      class="c1"
      height="12px"
      width="12px"
    >
      <svg
        class=""
        color="#BCBEC0"
        height="12px"
        viewBox="0 0 12 12"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 1H7v1h1V1zM6.5 6.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM6 3a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-1 0v2A.5.5 0 0 0 6 3zm3.5 5.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zm0-2h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM9 3a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-1 0v2A.5.5 0 0 0 9 3zm-.5 2.5h1a.5.5 0 0 0 0-1h-1a.5.5 0 0 0 0 1zm-3 0h1a.5.5 0 0 0 0-1h-1a.5.5 0 0 0 0 1zm-2 3h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM11 1h-1v1h1v9H1V2h1V1H1a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1zM3.5 6.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zM5 1H4v1h1V1zm1.5 7.5h-1a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1zm-4-3h1a.5.5 0 0 0 0-1h-1a.5.5 0 0 0 0 1zM3 3a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-1 0v2A.5.5 0 0 0 3 3z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </div>
    <input
      aria-label="startDate"
      autocomplete="off"
      class="c2"
      color="#001217"
      data-testid="DatepickerInput"
      font-family="Montserrat, sans-serif"
      font-size="14px"
      font-weight="600"
      id="startDate"
      placeholder="Placeholder"
      readonly=""
      tabindex="0"
      value=""
      width="100%"
    />
  </label>
</div>
`;

exports[`should render with value 1`] = `
.c0 {
  position: relative;
  border: 1px solid #d0d0d0;
  background: #ffffff;
  display: block;
  border-radius: 2px;
  margin: 0;
}

.c1 {
  background: #ffffff;
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  color: #001217;
  font-weight: 600;
  border: 0;
  width: 100%;
  min-height: 46px;
  box-shadow: none;
  cursor: pointer;
  box-sizing: border-box;
  outline: 0;
}

.c1::-webkit-input-placeholder {
  font-weight: 500;
  color: #929598;
}

.c1::-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

.c1:-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

<div>
  <label
    class="c0"
    display="block"
    for="startDate"
  >
    <input
      aria-label="startDate"
      autocomplete="off"
      class="c1"
      color="#001217"
      data-testid="DatepickerInput"
      font-family="Montserrat, sans-serif"
      font-size="14px"
      font-weight="600"
      id="startDate"
      placeholder="Placeholder"
      readonly=""
      tabindex="0"
      value="14/11/1992"
      width="100%"
    />
  </label>
</div>
`;

exports[`should render without icon 1`] = `
.c0 {
  position: relative;
  border: 1px solid #d0d0d0;
  background: #ffffff;
  display: block;
  border-radius: 2px;
  margin: 0;
}

.c1 {
  background: #ffffff;
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  color: #001217;
  font-weight: 600;
  border: 0;
  width: 100%;
  min-height: 46px;
  box-shadow: none;
  cursor: pointer;
  box-sizing: border-box;
  outline: 0;
}

.c1::-webkit-input-placeholder {
  font-weight: 500;
  color: #929598;
}

.c1::-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

.c1:-moz-placeholder {
  font-weight: 500;
  color: #929598;
}

<div>
  <label
    class="c0"
    display="block"
    for="startDate"
  >
    <input
      aria-label="startDate"
      autocomplete="off"
      class="c1"
      color="#001217"
      data-testid="DatepickerInput"
      font-family="Montserrat, sans-serif"
      font-size="14px"
      font-weight="600"
      id="startDate"
      placeholder="Placeholder"
      readonly=""
      tabindex="0"
      value=""
      width="100%"
    />
  </label>
</div>
`;
